### PR TITLE
Personal cards trackers

### PR DESCRIPTION
**Pull request by iravisankar**
Following trackers are added for personal cards
- personal cards page opened
- new card linked on personal cards
- card deleted on personal cards
- new expense created from personal card txn
- expense matched created from personal card txn
- expense matched created from personal card txn
- txns hidden on perosnal cards
- txns hidden on perosnal cards

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -500,7 +500,11 @@ export class AppComponent implements OnInit {
       },
       {
         title: 'Personal Cards',
-        isVisible: environment.ROOT_URL === 'https://staging.fyle.tech',
+        isVisible:
+          orgSettings.org_personal_cards_settings.allowed &&
+          orgSettings.org_personal_cards_settings.enabled &&
+          orgUserSettings.personal_cards_settings?.enabled &&
+          orgUserSettings.personal_cards_settings?.personal_cards_data_aggregator.length > 0,
```
Comment (abhishek1234321) [inline src/app/app.component.ts:507]: is this correct? webapp doesn't have this

File: src/app/core/models/org_user_settings.model.ts [hunk 2]
```diff
@@ -121,6 +121,10 @@ export interface ExpenseFormAutofills {
   allowed: boolean;
   enabled: boolean;
 }
+export interface personalCardsSettings {
```
Comment (abhishek1234321) [inline src/app/core/models/org_user_settings.model.ts:124]: `PersonalCardsSettings`

File: src/app/core/services/tracking.service.ts [hunk 3]
```diff
@@ -913,4 +913,37 @@ export class TrackingService {
   onSettingsToggle(properties) {
     this.eventTrack('Toggle Setting', properties);
   }
+
+  //Personal Cards
+  personalCardsViewed(properties = {}) {
+    this.eventTrack('personal cards page opened', properties);
```
Comment (abhishek1234321) [inline src/app/core/services/tracking.service.ts:919]: We use Title Case everywhere no in events?

Comment (abhishek1234321) [reply src/app/core/services/tracking.service.ts:919]: also, I don't think we need to specify `personal cards` - won't this appear in the event itself by default?

File: src/app/core/services/tracking.service.ts [hunk 4]
```diff
@@ -913,4 +913,37 @@ export class TrackingService {
...
+  oldExpensematchedFromPersonalCard(properties = {}) {
+    this.eventTrack('expense matched created from personal card txn', properties);
+  }
+
+  unmatchedExpensesFromPersonalCard(properties = {}) {
+    this.eventTrack('expense matched created from personal card txn', properties);
+  }
+
+  transactionsHiddenOnPersonalCards(properties = {}) {
+    this.eventTrack('txns hiden on perosnal cards', properties);
```
Comment (abhishek1234321) [inline src/app/core/services/tracking.service.ts:943]: `personal`

File: src/app/core/services/tracking.service.ts [hunk 5]
```diff
@@ -913,4 +913,37 @@ export class TrackingService {
...
+  }
+
+  transactionsFetchedOnPersonalCards(properties = {}) {
+    this.eventTrack('txns hiden on perosnal cards', properties);
```
Comment (abhishek1234321) [inline src/app/core/services/tracking.service.ts:947]: `transactions hidden`

**Review by abhishek1234321 — CHANGES_REQUESTED** [review]

File: src/app/app.component.ts [hunk 6]
```diff
@@ -500,7 +500,11 @@ export class AppComponent implements OnInit {
       },
       {
         title: 'Personal Cards',
-        isVisible: environment.ROOT_URL === 'https://staging.fyle.tech',
+        isVisible:
+          orgSettings.org_personal_cards_settings.allowed &&
+          orgSettings.org_personal_cards_settings.enabled &&
+          orgUserSettings.personal_cards_settings?.enabled &&
+          orgUserSettings.personal_cards_settings?.personal_cards_data_aggregator === 'YODLEE',
```
Comment (iravisankar) [inline src/app/app.component.ts:507]: kulasekar confirmed this...

File: src/app/core/services/tracking.service.ts [hunk 7]
```diff
@@ -913,4 +913,37 @@ export class TrackingService {
   onSettingsToggle(properties) {
     this.eventTrack('Toggle Setting', properties);
   }
+
+  //Personal Cards
+  personalCardsViewed(properties = {}) {
+    this.eventTrack('Personal cards page opened', properties);
```
Comment (iravisankar) [inline src/app/core/services/tracking.service.ts:919]: Mentioned `Personal cards ` because it helps to identify while creating a report on mixpanel.

**Review by abhishek1234321 — APPROVED** [review]

**Pull request merged**